applications/nrf_desktop/doc/dvfs.rst

```diff
@@ -0,0 +1,104 @@
+    If timeout is set to 0 the <STATE> will not cease requesting its active frequency until receiving off event.
+
+Each change request is retried if it fails due to known reason.
+The module will retry the request up to the number of times specified by :option:`CONFIG_DESKTOP_DVFS_RETRY_COUNT`.
+Error counter is only zeroed when preforming correct DVFS change request.
+Failing frequency change requests can be caused by another frequency change request being in progress on sysctrl - error ``-EBUSY`` or by the nrfs DVFS service being not yet initialized - error ``-EAGAIN``.
+If the frequency change request is in progress, the module will retry the request after a timeout specified by :option:`CONFIG_DESKTOP_DVFS_RETRY_BUSY_TIMEOUT_MS`.
+If the nrfs service DVFS is not yet initialized, the nRF Desktop DVFS module will retry the request after a timeout specified by :option:`CONFIG_DESKTOP_DVFS_RETRY_INIT_TIMEOUT_MS`.
+If the DVFS frequency request returns an error different than ``-EBUSY`` or ``-EAGAIN``, the module will report ``MODULE_STATE_ERROR`` and set ``module_state`` to ``STATE_ERROR``.
+nRF Desktop DVFS module will also report MODULE_STATE_ERROR and set ``module_state`` to ``STATE_ERROR`` if the number of retries exceeds the value specified by :option:`CONFIG_DESKTOP_DVFS_RETRY_COUNT`.
+In ``module_state`` in ``STATE_ERROR`` the module will not change the frequency and voltage.
+
+* :ref:`CONFIG_DESKTOP_DVFS_RETRY_BUSY_TIMEOUT_MS <config_desktop_app_options>` - Timeout in milliseconds specifying time after which nRF Desktop DVFS module will retry DVFS frequency change.
+   This timeout is applied in case another DVFS change request is still in progress which causes the current request to fail.
+   The default value is 1 ms.
+* :ref:`CONFIG_DESKTOP_DVFS_RETRY_INIT_TIMEOUT_MS <config_desktop_app_options>` - Timeout in milliseconds specifying time after which nRF Desktop DVFS module will retry DVFS frequency change.
+   This timeout is applied in case DVFS is not yet initialized which causes the current request to fail.
+   The default value is 500 ms.
+* :ref:`CONFIG_DESKTOP_DVFS_RETRY_COUNT <config_desktop_app_options>` - Number of retries of DVFS frequency change after which nRF Desktop DVFS module will report MODULE_STATE_ERROR.
+   The default value is 5.
+
+Implementation details
+**********************
+
+The nRF Desktop Dynamic Voltage and Frequency Scaling (DVFS) module has two state tracing entities.
+
+First ``module_state`` is used to track the state of the nRF Desktop DVFS module.
+It indicates if the module is ready or if an error has occurred.
+
+Second ``dfvs_requests_state_bitmask`` is used to track separately the state of each DVFS state.
+Each DVFS state is represented by a bit in the bitmask.
+If the bit is set, the state is active.
+The module listens for events that are sent and applies state activity according to the event.
```

Comment:
`The module listens for Event Manager events related to a given state and updates state accordingly`.